Fix Issues

dan-tang-ssd opened the pull request:
This PR is submitted to fix issue https://github.com/stats4sd/holpa-platform/issues/300

It needs to be reviewed #together with submodule filament-team-management PR https://github.com/stats4sd/filament-team-management/pull/44

It is not yet ready for review. It is submitted for progress update.
More items to be added.

---

It contains below changes:
 - [x] A05, Show Program Admin Panel menu item when login as Program Admin
 - [x] A19, FarmResource, disable New Farm button if there is no location level with farm
 - [x] A24, Contextualise choice lists, error occurred. for real HOLPA forms, seeder file, make correction to set choice_lists first record column is_localisable to 1
 - [x] B11, Submodule update - Register new team, set team name field as compulsory; change team name field label but failed (This issue is resolved in Filament 4)
 - [x] B35, Setup soft delete for Team model; My Teams page > click "Delete" button, redirect to /app survey dashboard. User will be prompted to create a new team if the deleted team is the last team
 - [x] B35, TODO: Admin panel, add a filament resource to show the soft deleted teams
 - [x] B37, My Account page, re-arrange which password field is required when changing password; if new password is not entered, it should not call ODK link package to update login password in ODK central
 - [x] B37, TODO: My Account page, clear current password field value after update
 - [x] D02, Survey Country & Languages, add validation; select the newly created country
 - [x] E09, Survey Locations resource > Locations relation manager, add new location, add validation to add unique id only
 - [x] E10, Import locations with same unique ID without error. Checked program source code. The import location program upsert location details. It is intended to update location with same unique ID instead of showing error message.
 - [x] E14 & E15, Farms page, import location and farms button, fix issue for duplicate farms records created after import
 - [x] Farms page, import location farms list button, and import farm list button, do not create new farms record if farms record with same unique code is already existed for the same team
 - [x] E17, Survey Locations > Farms resource > create / edit farm, add validation to create unique code (team_code) per team
 - [x] E18, Context Questions page, add breadcrumb; add workaround to avoid error

---

Added validations:
 - [x] Survey Translations page > Language > Add New button, translation name, change to required
 - [x] Location Levels page > New location level button > parent location level, exclude the current location level record, to prevent self-referencing loops
 - [x] Farms page, Import location and farm list button and Import farm list button, disable if there is no location level with farms
 - [x] Location page, remove unique constraint for column locations.code; location name and location code are unique per team now
 - [x] Location page, fix import locations after removing unique constraint of column locations.code. Check location record existence then create record if location code not existed for this team yet

---

Questions:
1. Column locations.code has unique constraint in database level. That means location code is unique among all teams. Should we remove the unique constraint to allow location code to be unique per team?

dave-mills commented:
> Question: I updated some program files in submodule, but it does not take effect in my local env.

I took a quick look. The Page classes that are registered into the AppPanelProvider from the team management package are the ones from the vendor folder. But in my local environment, for some reason the package folder in vendor is _not_ symlinked to `packages/`. (The ODK Link package is, but the other is not), which will explain why your changes in `packages/` are not taking effect. 

For me, deleting the vendor folder and re-running composer install worked - it was recreated as a symlink.

dan-tang-ssd commented:
> > Question: I updated some program files in submodule, but it does not take effect in my local env.
> 
> I took a quick look. The Page classes that are registered into the AppPanelProvider from the team management package are the ones from the vendor folder. But in my local environment, for some reason the package folder in vendor is _not_ symlinked to `packages/`. (The ODK Link package is, but the other is not), which will explain why your changes in `packages/` are not taking effect.
> 
> For me, deleting the vendor folder and re-running composer install worked - it was recreated as a symlink.

Thank you Dave.

I deleted vendor folder then run command "composer install".
But it seems still not a symlink for submodule filament-team-management... Any idea?

---

Screen shots:

BEFORE:
<img width="819" height="149" alt="image" src="https://github.com/user-attachments/assets/65434a21-854d-4bc2-8e53-dc877fafea4e" />

AFTER:
<img width="819" height="149" alt="image" src="https://github.com/user-attachments/assets/e91178f1-24ba-481c-8432-47a2a8f22c55" />

---

<img width="998" height="751" alt="image" src="https://github.com/user-attachments/assets/66c3d901-9272-4ec6-b7d8-3ef70c3cc416" />

<img width="1920" height="982" alt="image" src="https://github.com/user-attachments/assets/81e9401a-9e58-4cc0-b47f-fd9de95547bb" />

dave-mills commented:
Ah, maybe also try deleting the composer.lock file.
